### PR TITLE
[ENG-886] add newrelic error notice for Integration and connection exceptions

### PR DESCRIPTION
**Pull request by scottshipman**

### CODE REVIEW

File: Integration/ClientIntegration.php [hunk 1]
```diff
@@ -800,12 +801,31 @@ private function handleException(\Exception $exception)
             if ($errorData) {
                 $this->setLogs($errorData, $exception->getStatType());
             }
+        } elseif ($exception instanceof ConnectException) {
+            if (function_exists('newrelic_notice_error')) {
+                call_user_func(
+                    'newrelic_notice_error',
+                    'ContactClient Connection Error: '.$this->contactClient->getName().' ['.$this->contactClient->getId(
```
Comment (dean-martin) [inline Integration/ClientIntegration.php]: Double quotes would make this easier to read imo 
`"ContactClient Connection Error: {$this->contactClient->getName()} [{$this->contactClient->getId()}] in Campaign: {$this->campaign->getName()} [{$this->campaign->getId()}]",`

File: Integration/ClientIntegration.php [hunk 2]
```diff
@@ -800,12 +801,31 @@ private function handleException(\Exception $exception)
...
+                    ).'] in Campaign: '.$this->campaign->getName().' ['.$this->campaign->getId().']',
+                    $exception
+                );
+                $this->logIntegrationError($exception, $this->contact);
```
Comment (heathdutton) [inline Integration/ClientIntegration.php]: Line can be dropped.

File: Integration/ClientIntegration.php [hunk 3]
```diff
@@ -800,12 +801,31 @@ private function handleException(\Exception $exception)
...
-            $this->logIntegrationError($exception, $this->contact);
+            if (function_exists('newrelic_notice_error')) {
+                call_user_func(
+                    'newrelic_notice_error',
+                    'ContactClient Integration Error: '.$this->contactClient->getName(
+                    ).' ['.$this->contactClient->getId().'] in Campaign: '.$this->campaign->getName(
+                    ).' ['.$this->campaign->getId().']',
+                    $exception
+                );
+                // Unexpected issue with the Client plugin.
```
Comment (heathdutton) [inline Integration/ClientIntegration.php]: The following lines need to be one } up in the hierarchy.

File: Integration/ClientIntegration.php (hunk 2 above)
Comment (scottshipman) [reply Integration/ClientIntegration.php]: fixed

File: Integration/ClientIntegration.php (hunk 1 above)
Comment (heathdutton) [reply Integration/ClientIntegration.php]: I feel like this is debatable. Either could be more readible depending on your IDE and personal training. There's no code-style enforcing either convention so I wouldn't reject a PR for it.

**Pull request merged**